File: 2016/finale/miglia/att/miglia.c
#include <stdio.h>
#include <assert.h>

#define MAXM 10000

int vola(int K, int N, int M, int da[], int a[], int V[]) {
    // Mettete qui il codice della soluzione
    return 42;
}


int da[MAXM];
int a[MAXM];
int V[MAXM];

int main() {
    FILE *fr, *fw;
    int K, N, M, i;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(3 == fscanf(fr, "%d %d %d", &K, &N, &M));
    for(i=0; i<M; i++)
        assert(3 == fscanf(fr, "%d %d %d", &da[i], &a[i], &V[i]));

    fprintf(fw, "%d\n", vola(K, N, M, da, a, V));
    fclose(fr);
    fclose(fw);
    return 0;
}

File: ed2/finale/miglia/att/miglia.pas
function vola(K, N, M: longint; var da, a, V: array of longint): longint;
begin
    (* Mettete qui il codice della soluzione *)
    vola := 42;
end;


const
    MAXM = 10000;
var
    K,N,M,i : longint;
    da      : array[0..MAXM-1] of longint;
    a       : array[0..MAXM-1] of longint;
    V       : array[0..MAXM-1] of longint;
    fr, fw  : text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, K, N, M);
    for i:=0 to M-1 do
        readln(fr, da[i], a[i], V[i]);

    writeln(fw, vola(K, N, M, da, a, V));
    close(fr);
    close(fw);
end.
